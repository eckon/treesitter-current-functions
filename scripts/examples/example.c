#include <stdio.h>

#ifdef COMPILER_OPTION
void foo() {}
#endif

#if MAGIC_NUMBER > 1
void bar() {}
#else
void baz() {}
#endif

struct student
{
   char name[50];
   int height;
};

int main(){
}

bool8 sub_8039880(void)
{
  return (CountMailType(WONDER_MAIL_TYPE_SOS) != 0 || CountMailType(WONDER_MAIL_TYPE_OKD) != 0 || sub_8011C1C() == 2);
}
