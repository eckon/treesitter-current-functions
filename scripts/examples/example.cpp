#include <cstdint>
#include <utility>

/*
 * Test 1: Basics, method definitions outside the class definition. Also testing basic
 *          functions with primitive, pointer, and reference return types.
 */

class test_1_out {

private:
    int a, b;

public:
    std::pair<int, int> get_ab() const;
    test_1_out(int _a, int _b);
    ~test_1_out();
};
test_1_out *test_1_out_inst = nullptr;

test_1_out::test_1_out(int _a, int _b)   { a = _a; b = _b; }
test_1_out::~test_1_out()                {  }
std::pair<int, int> test_1_out::get_ab() const { return { a, b }; }

test_1_out get_test_1_out_inst_reg() {
    return *test_1_out_inst;
}
test_1_out *get_test_1_out_inst_ptr() {
    return test_1_out_inst;
}
test_1_out &get_test_1_out_inst_ref() {
   return *test_1_out_inst;
}


/*
 * Test 2: Templated functions outside of class definitions.
 */

template <typename T>
class test_2_out {

private:
    T a, b;

public:
    std::pair<T, T> get_ab() const;
    test_2_out(T _a, T _b);
};
test_2_out<std::uint64_t> *test_2_out_inst = nullptr;

template <typename T> test_2_out<T>::test_2_out(T _a, T _b) { a = _a; b = _b; }
template <typename T> std::pair<T, T> test_2_out<T>::get_ab() const { return { a, b }; }

test_2_out<std::uint64_t> get_test_2_out_inst_reg() {
    return *test_2_out_inst;
}
test_2_out<std::uint64_t> *get_test_2_out_inst_ptr() {
    return test_2_out_inst;
}
test_2_out<std::uint64_t> &get_test_2_out_inst_ref() {
    return *test_2_out_inst;
}


/*
 * Test 3: Operator overloading.
 */

struct test_3_out {
    std::uint64_t x, y, z;
};

test_3_out operator+(const test_3_out &a, const test_3_out &b) {
    return { a.x + b.x, a.y + b.y, a.z + b.z };
}

test_3_out operator-(const test_3_out &a, const test_3_out &b) {
    return { a.x - b.x, a.y - b.y, a.z - b.z };
}


/*
 * Test 4: Namespaces
 */

namespace test_4 {

    struct test_4a {
        void foo();
    };

    struct test_4b {
        void foo();
    };
    void test_4b::foo() { /* ... */ }

    void foo_a() { /* ... */ }
    void foo_b();
}

void test_4::test_4a::foo() { /* ... */ }
void test_4::foo_b()        { /* ... */ }


/* TODO: Functions declared inside a class as follows still don't appear

   class a {
       void foo() { ... }
   };

*/

