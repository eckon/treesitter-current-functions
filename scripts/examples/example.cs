namespace Example.Model
{
    public class ExampleManager
    {
        public ExampleManager()
        {
        }

        public async Task<foo> Start(int number, string test)
        {
        }

        private string End()
        {
        }
    }
}
