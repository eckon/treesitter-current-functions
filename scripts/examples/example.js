function multiline(
  a, b
) {}

function inline() {}

function inline2() {}

function inline3() {}

const arrowFunction = () => {};

class FirstClass {
  constructor(a) {}

  firstMethod() {}

  secoundMethod(a, v, c) {}
}

export class SecoundClass {
  thirdMethod() {}
}
