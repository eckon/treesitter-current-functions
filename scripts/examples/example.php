<?php

function firstFunction() {}

function secoundFunction() {}

class FirstClass {
    function firstMethod() {}
}
