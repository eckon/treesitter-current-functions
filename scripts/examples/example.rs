
#[allow(dead_code)]
struct Test1 {
    foo: usize
}
impl Test1 {
    fn new() -> Self {
        Test1 { foo: 0 }
    }
}

#[allow(dead_code)]
enum Test2 {
    Item1,
    Item2,
    Item3,
}
impl Test2 {
    fn index(&self) -> usize {
        match self {
            Test2::Item1 => 0,
            Test2::Item2 => 1,
            Test2::Item3 => 2,
        }
    }
}

#[allow(unused_variables)]
fn main() {
    let test1 = Test1::new();
    let test2 = Test2::Item1;
    test2.index();
}
