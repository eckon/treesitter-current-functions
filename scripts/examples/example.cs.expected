{ {
    function_name = "Example.Mod...ExampleManager > ExampleManager",
    line_number = 5
  }, {
    function_name = "Example.Model > ExampleManager > Start",
    line_number = 9
  }, {
    function_name = "Example.Model > ExampleManager > End",
    line_number = 13
  } }
